src/LibChorus/VcsDrivers/Mercurial/HgRepository.cs

```diff
@@ -1475,10 +1471,28 @@ private static void SetExtensions(IniDocument doc, IEnumerable<KeyValuePair<stri
 			}
 		}
 
-		internal static bool CheckExtensions(IniDocument doc, IEnumerable<KeyValuePair<string, string>> extensionDeclarations)
+		/// <summary/>
+		/// <returns>
+		/// Returns true if the extensions in the repo contain only the extensions declared in the given map.
+		///</returns>
+		internal static bool CheckExtensions(IniDocument doc, Dictionary<string, string> extensionDeclarations)
 		{
-			var section = doc.Sections.GetOrCreate("extensions");
-			return extensionDeclarations.All(pair => section.GetValue(pair.Key) == pair.Value);
```

Comment:
I rather like doing it in one line with All (would be two lines now, of course).